### ISSUE TITLE
Tactor Sourcing/List of potential products

### ISSUE THREAD

**Issue by danteoh** [RChadwick7/ParkinsonsGloves#2]
Are we able to source the EA Tactors used and specs to drive them with an Arduino/other PWM hardware?

Specifications are covered here:

https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8055937/bin/Data_Sheet_1.pdf

These are the Vybronics motors I used to get the prototype across the line but if I can move ov er to the EA tactors that'd be preferable.

https://www.digikey.com/en/products/detail/vybronics-inc/VW0625AB001G/9974285?utm_adgroup=Motors%20-%20AC%2C%20DC&utm_source=google&utm_medium=cpc&utm_campaign=Shopping_Product_Motors%2C%20Solenoids%2C%20Driver%20Boards%2FModules_NEW&utm_term=&utm_content=Motors%20-%20AC%2C%20DC&gclid=CjwKCAiAkrWdBhBkEiwAZ9cdcIa5yR9HryC2NRqx0emIQMeRBLrPV4YbukfdilMS01gTFgao_otIDhoC0m8QAvD_BwE

**Comment by danteoh** [comment]
Also.. they list no sales folks on LInkedin?

**Comment by Beartech** [comment]
I contacted the company EA in Florida. They will sell you the tactors they are using and a programmable unit (they can't sell you the unit that was developed for the study as it is now considered an FDA controlled device. Here are the data sheets they sent me:

[C-MF Product Data Sheet 01_21.pdf](https://github.com/RChadwick7/ParkinsonsGloves/files/10377154/C-MF.Product.Data.Sheet.01_21.pdf)
[Universal Controller Product Data Sheet 03_20.pdf](https://github.com/RChadwick7/ParkinsonsGloves/files/10377155/Universal.Controller.Product.Data.Sheet.03_20.pdf)

From the email:
Brian Altenbernd <baltenbernd@eaiinfo.com> Wed, Dec 21, 2022 at 11:12 AM

Hi Andy,
Pricing is below:
C-MF Tactor @ $360 each (at qty 8)

Universal Controller @ $2,350 (additional $290 for Bluetooth option)
Price does not include shipping cost

Current Delivery Estimate: 4 weeks
We recommend our Universal Controller to drive our tactors due to its capabilities and simplicity. The Universal Controller can operate up to 8 tactors and includes our TDK software. Please note that is not the same controller used with the gloves.
I have provided the data sheets for the C-MF Tactor and Universal Controller for your review. Please let me know if you have any additional questions.

**Comment by danteoh** [comment]
Haha I just reached out on LinkedIn to him figuring him the closest sales person they had before you commented.

**Comment by Beartech** [comment]
I have two of the TDK piezo actuators I got from DigiKey. They are just the bare Piezo element. They would need a controller that would produce a +/- 60V signal at 250Hz. There were some easy to use board from a company (like Adafruit, but can't remember the name) but they are not in stock anymore. But those boards did quite a bit on something the size of a postage stamp, like remembering different vibration patterns etc. I think they can easily be driven by a small amp circuit of some kind.  I also just ordered some tiny speakers, like the ones in cell phones with the intention of trying the floating coil design others have proposed. 

The piezo elements need a certain circuit to operate. A speaker can easily be driven by an analog sound signal. I'm going to try cutting away most of the cone of the speaker to get rid of air resistance and then mount some sort of stud in the middle of the cone to touch the finger. The reason piezo elements were used instead of eccentric motors/LRAs is that the vibration periods are very short, and they need to start and stop very quickly so that they don't overlap on finger stimulation. Piezo elements are very good at this, probably due to the high voltages involved. But a speaker should also be pretty good at it since the voice coil moves very easily. I suppose if you have an LRA that can get moving and stop very quickly, it might work.

**Comment by danteoh** [comment]
So have rotational mass vibrators been ruled out by y'all?

**Comment by danteoh** [comment]
![image](https://user-images.githubusercontent.com/1686515/211419830-0bfec1dd-3f6a-437f-b838-f888a2a37c24.jpeg)

These are my V1 finger pads that I 3D printed.

**Comment by HackyDev** [comment]
> They would need a controller that would produce a +/- 60V signal at 250Hz. 

Hey! Check out the [DRV8662EVM](https://www.ti.com/tool/DRV8662EVM) developed by Texas Instruments. This driver can work with a wide range of piezoelectric elements, including those commonly found in fire alarms. By connecting these inexpensive piezo discs to the DRV8662EVM, you can produce significant vibrations. Although it is not entirely clear how long these elements can withstand the applied voltage. To get some understanding of the device's uses and applications, I recommend watching this [video](https://www.youtube.com/watch?v=egLA0GgF3xw).

**Comment by HackyDev** [comment]
> So have rotational mass vibrators been ruled out by y'all?

One of the major challenges I encountered was the slow acceleration of the rotational mass vibrators. Due to their high inertia, it takes a significant amount of time for the motors to reach their maximum speed. My motors take 300ms to reach their peak speed which is significantly longer than the specifications' requirement of less than 2ms.  Another major issue I found is that the vibrations produced by the rotational mass vibrators are excessively strong, when they should be barely perceptible.

**Comment by danteoh** [comment]
@HackyDev yeah one thing I haven't been able to compare apples to apples is amplitude of the vibrations from the EMR versus the Tactors from EAI. we could reduce that with rubber insulators potentially? 

The vybronics are pretty fast reaction time.

**Comment by HackyDev** [comment]
> @HackyDev We could reduce that with rubber insulators potentially?

I couldn't find a way to effectively lower the amplitude and remove unnecessary vibrations in a wearable device, so I decided to create two stationary pads and a contraption for each finger, as shown in the image below. The bottom part of the contraption is intended to be placed in a container filled with cement/epoxy or something like that. The best part of this design it's very cheap, but will it be effective? Probably not.  

![Untitled](https://user-images.githubusercontent.com/70698428/211670570-bd474680-469b-4a1b-80e8-e81000745154.jpg)

**Comment by RChadwick7** [comment]
Wow, Awesome work! I can't wait to show off what I have, although it's ugly compared to these. As for ERM, I think running them with a full H driver would help. A chip like DRV8833 or L9110S will let you short the motor when you aren't driving it, which should stop the motor much faster. As for 300ms startup, that's a long time! Can your power supply source enough current?
